libafl_qemu/librasan/asan/src/mem.rs

```diff
@@ -34,107 +33,24 @@ static GLOBAL_ALLOCATOR: DlmallocBackend<Mmap> = DlmallocBackend::new(PAGE_SIZE)
 static GLOBAL_ALLOCATOR: baby_mimalloc::MimallocMutexWrapper<DlmallocBackend<Mmap>> =
     baby_mimalloc::MimallocMutexWrapper::with_os_allocator(DlmallocBackend::new(PAGE_SIZE));
 
-#[unsafe(no_mangle)]
-pub unsafe extern "C" fn memmove(dest: *mut u8, src: *const u8, count: usize) {
-    let src_slice = unsafe { from_raw_parts(src, count) };
-    let dest_slice = unsafe { from_raw_parts_mut(dest, count) };
-
-    if src < dest {
-        #[allow(clippy::manual_memcpy)]
-        for i in 0..count {
-            let idx = count - 1 - i;
-            dest_slice[idx] = src_slice[idx];
-        }
-    } else {
-        #[allow(clippy::manual_memcpy)]
-        for i in 0..count {
-            dest_slice[i] = src_slice[i];
-        }
-    }
-}
-
-#[unsafe(no_mangle)]
-pub unsafe extern "C" fn memcpy(dest: *mut u8, src: *const u8, count: usize) {
-    let src_slice = unsafe { from_raw_parts(src, count) };
-    let dest_slice = unsafe { from_raw_parts_mut(dest, count) };
-    #[allow(clippy::manual_memcpy)]
-    for i in 0..count {
-        dest_slice[i] = src_slice[i];
-    }
-}
-
-#[unsafe(no_mangle)]
-pub unsafe extern "C" fn memset(dest: *mut u8, value: u8, count: usize) {
-    unsafe {
-        let mut cursor = dest;
-        let word_value = match value {
-            u8::MIN => Some(usize::MIN),
-            u8::MAX => Some(usize::MAX),
-            _ => None,
-        };
-
-        if let Some(word_value) = word_value {
-            let num_words = count / size_of::<usize>();
-            for _ in 0..num_words {
-                *(cursor as *mut usize) = word_value;
-                cursor = cursor.wrapping_add(size_of::<usize>());
-            }
-
-            let num_bytes = count % size_of::<usize>();
-            for _ in 0..num_bytes {
-                *cursor = value;
-                cursor = cursor.wrapping_add(1);
-            }
-        } else {
-            for _ in 0..count {
-                *cursor = value;
-                cursor = cursor.wrapping_add(1);
-            }
-        }
-    }
-}
-
-#[unsafe(no_mangle)]
-pub unsafe extern "C" fn memcmp(ptr1: *const u8, ptr2: *const u8, count: usize) -> i32 {
-    let slice1 = unsafe { from_raw_parts(ptr1, count) };
-    let slice2 = unsafe { from_raw_parts(ptr2, count) };
-
-    for i in 0..count {
-        match slice1[i].cmp(&slice2[i]) {
-            Ordering::Equal => (),
-            Ordering::Less => return -1,
-            Ordering::Greater => return 1,
-        }
-    }
-
-    0
-}
-
-#[unsafe(no_mangle)]
-pub unsafe extern "C" fn bcmp(ptr1: *const u8, ptr2: *const u8, count: usize) -> i32 {
-    let slice1 = unsafe { from_raw_parts(ptr1, count) };
-    let slice2 = unsafe { from_raw_parts(ptr2, count) };
+/*
+ * The functions are only called from our C code, but we need to tell Rust that
+ * we still need it even though it isn't referenced from rust.
+ */
```

Comment:
I'd change this comment to something like the following:

> The Rust standard library expects these functions from libc to be present when building with `no_std` on an `*-gnu` target. We include implementations for these functions from a crate, but we still need to create a reference here to ensure the object files compiled from that crate end up in our static library.

(At least, I'm assuming the second sentence is the reason for this requirement.)